src/file.ts

```diff
@@ -1802,7 +1802,9 @@ class File extends ServiceObject<File> {
       const configDir = xdgBasedir.config || os.tmpdir();
 
       fs.access(configDir, fs.constants.W_OK, err => {
-        if (err) {
+        if (err && fs.existsSync(configDir)) {
```

Comment:
Generally you want to avoid calling `Sync` methods inside of production level code paths.  They block the node.js event loop, and file system access is slow.  If we wanted to keep down this path, I think we'd have to make 2 distinct calls to `fs.access` with varying flags:  one with `R_OK` and one with `W_OK`.  

That having been said ...   the pattern of checking for a given directory or file before starting a write is generally frowned upon.  This is specifically called out in the `fs` docs:

> Do not use `fs.access()` to check for the accessibility of a file before calling `fs.open()`, `fs.readFile()` or `fs.writeFile()`. Doing so introduces a race condition, since other processes may change the file's state between the two calls. Instead, user code should open/read/write the file directly and handle the error raised if the file is not accessible.

From here:
https://nodejs.org/api/fs.html#fs_fs_access_path_mode_callback

Should we instead handle errors that come back directly from `startSimpleUpload_`, and give a good error message based on the resulting stream?

Reply:
Adding to the first point - if we decide to keep going down the pre-check route, it's possible you can do a single `fs.stat` and do bit math to play with the resulting mode - I'm out of my depth here though (@bcoe may know more)